pytorch_lightning/plugins/ddp_plugin.py

```diff
@@ -91,23 +92,23 @@ def init_ddp_connection(
                 torch_backend, rank=global_rank, world_size=world_size
             )
 
-    def on_before_forward(self, model: LightningModule, *args):
+    def on_before_forward(self, model: LightningModule, *batch):
         """
         Override to handle custom input to device logic. For DDP, no logic is required as this is handled internally
         within the DDP wrapper.
 
         Example::
 
-            def on_before_forward(self, model, *args):
-                batch, batch_idx = args
+            def on_before_forward(self, model, *batch):
+                batch, batch_idx = batch
                 return batch.to(model.device)
 
         Args:
-            args: Inputs to the model.
+            batch: Inputs to the model.
             model: Model to train.
-        Returns: args moved to correct device if needed.
+        Returns: batch moved to correct device if needed.
         """
-        return args
+        return model.transfer_batch_to_device(batch, model.device)
```

Comment:
not very sure about passing this arg as it came from * so it can contain a few args, but basically it shall be a tuple, right?

Reply:
It should work fine.